validation-test/stdlib/CollectionTypeInference.swift

```diff
@@ -0,0 +1,29 @@
+// RUN: %target-parse-verify-swift
+
+struct FatalIndex : Comparable { }
```

Comment:
Could you add a comment that explains what the test is checking (that you can supply a minimal list of requirements).